### PR TITLE
Add pin, unpin and setpinpath for maps in libbpfgo

### PR DESCRIPTION
**Pull request by ogozman**
Addid pinning for maps in libbpfgo

### CODE REVIEW

**Review by itaysk — CHANGES_REQUESTED** [review]
please free the allocated strings. you can take example from other functions that use CString

File: libbpfgo/libbpfgo.go [hunk 1]
```diff
@@ -302,6 +302,39 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
...
+	path := C.CString(pinning)
+	bpfMap := C.bpf_object__find_map_by_name(b.module.obj, cs)
+	errC := C.bpf_map__unpin(bpfMap, path)
+	if errC != 0 {
+		return fmt.Errorf("Failed unpin path for map %s", b.name)
+	}
+	return nil
+}
+
+func (b *BPFMap) SetPinPath(pinning string) error {
```
Comment (itaysk) [inline libbpfgo/libbpfgo.go:327]: looks like this does the same as unpin? what did I miss here?

**Review by itaysk — COMMENTED** [review]
can you please rename "pinning" to meaningful name, like "

File: libbpfgo/libbpfgo.go (hunk 1 above)
Comment (ogozman) [reply libbpfgo/libbpfgo.go:327]: You did not miss anything, it was my fault, i did not push the change after testing

Comment (ogozman) [reply libbpfgo/libbpfgo.go:327]: Would "pinPath" be a suitable name?

**Review by yanivagman — APPROVED** [review]

**Comment by grantseltzer** [comment]
This is really good to have, thanks @ogozman!

File: libbpfgo/libbpfgo.go (hunk 1 above)
Comment (ogozman) [reply libbpfgo/libbpfgo.go]: I have freed pointer and changed the "pinning" to "pinPath"

**Review by itaysk — APPROVED** [review]

**Pull request merged**